static/js/services/enketo.js

```diff
@@ -109,7 +116,18 @@ angular.module('inboxServices').service('Enketo',
           })
           .then(transformXml);
       }
-      return xmlCache[id][language];
+      return $q.resolve()
+        .then(function() {
+          return xmlCache[id][language];
+        })
+        .then(function(doc) {
+          // clone doc to avoid leaking of data between instances of a form
+          return {
+            html: doc.html.clone(),
+            model: doc.model,
+            hasContactSummary: doc.hasContactSummary
```

Comment:
👍 if there's an alternative to `doc` that could be an improvement - i assumed it was a doc stored in couch.